OSOE-935: Migrating tests, Lombiq.UITestingToolbox and Lombiq.Tests to xUnit v3

Piedone opened the pull request:
[OSOE-935](https://lombiq.atlassian.net/browse/OSOE-935)

[OSOE-935]: https://lombiq.atlassian.net/browse/OSOE-935?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

Piedone committed:
Migrating tests to xUnit v3

Piedone committed:
Fixing analyzer violation

Piedone committed:
Implementing test cancellation in Lombiq.UITestingToolbox

Piedone committed:
NuGet publishes

Piedone committed:
Removing mistaken UI test project reference from web project

Piedone committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-935

# Conflicts:
#	tools/Lombiq.GitHub.Actions

Piedone committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-935

# Conflicts:
#	src/Modules/Lombiq.ChartJs
#	src/Modules/Lombiq.ContentEditors
#	src/Modules/Lombiq.DataTables
#	src/Modules/Lombiq.HelpfulExtensions
#	src/Modules/Lombiq.JsonEditor
#	src/Modules/Lombiq.Privacy
#	src/Modules/Lombiq.UIKit
#	src/Modules/Lombiq.VueJs
#	src/Modules/Lombiq.Walkthroughs
#	src/Themes/Lombiq.BaseTheme

Piedone committed:
Merging submodules

Piedone committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-935

Piedone committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-935

# Conflicts:
#	test/Lombiq.UITestingToolbox
#	tools/Lombiq.GitHub.Actions

Piedone committed:
Updating submodules to latest

Piedone committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-935

Piedone committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-935

# Conflicts:
#	src/Modules/Lombiq.DataTables
#	src/Modules/Lombiq.VueJs
#	src/Utilities/Lombiq.NodeJs.Extensions
#	test/Lombiq.UITestingToolbox

Piedone committed:
Removing leftover Xunit.Abstractions references

Piedone committed:
Fixing VS test discovery

Piedone committed:
Updating NuGetTest

Piedone committed:
Submodule test/Lombiq.UITestingToolbox updated

Submodule test/Lombiq.UITestingToolbox:
    > 6a0fac48 - Moving test timeout management to UITestExecutionSession
    > 3ca6fc68 - Making the Workflow test of BasicOrchardFeaturesTesting more reliable

Piedone committed:
Spelling

Piedone committed:
Updating NuGetTest

Piedone committed:
Removing debug code

Piedone committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-935

# Conflicts:
#	src/Libraries/Lombiq.HelpfulLibraries
#	test/Lombiq.UITestingToolbox

Piedone committed:
Merging submodules

Piedone committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-935

Piedone committed:
Making EnteringInteractiveModeShouldWait more reliable

Piedone committed:
Fixing submodule references

Piedone committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-935

# Conflicts:
#	src/Libraries/Lombiq.HelpfulLibraries
#	src/Libraries/Lombiq.OrchardCoreApiClient
#	src/Modules/Lombiq.LoginAsAnybody
#	tools/Lombiq.GitHub.Actions

Piedone committed:
More docs on remote tests

Piedone committed:
Spelling

Piedone committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-935

# Conflicts:
#	src/Modules/Lombiq.DataTables

Piedone committed:
Removing unneeded reference

Piedone committed:
Dummy change to kick off build

Piedone committed:
Revert "Dummy change to kick off build"

This reverts commit a3c19fbe45bab5c7fb2bda09dded1652c97a4c42.

Piedone committed:
Increasing timeout

Piedone committed:
Revert "Increasing timeout"

This reverts commit 9b3fe3bde04c83318bea7163bee01a06e1268e07.

Piedone committed:
Fixing that TestRunTimeoutShouldThrowAsync always created a test dump

Piedone committed:
Testing test-diagnostic-mode

Piedone committed:
Trying DOTNET_SYSTEM_NET_SOCKETS_INLINE_COMPLETIONS

Piedone committed:
Temp workflow

Piedone committed:
Only running Lombiq.Tests.UI.Samples tests

Piedone committed:
Trying larger machine

dministro reviewed (changes requested):
Merge from dev please.

Piedone committed:
Disabling HTML validation as a test

Piedone committed:
Merging submodules

Piedone committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-935

# Conflicts:
#	.github/workflows/build-and-test.yml
#	src/Libraries/Lombiq.OrchardCoreApiClient
#	src/Modules/Lombiq.HelpfulExtensions
#	src/Modules/Lombiq.Hosting.Tenants
#	src/Modules/Lombiq.LoginAsAnybody
#	src/Themes/Lombiq.Hosting.MediaTheme

Piedone committed:
Fixing build error

Piedone committed:
UITT clean-up

Piedone committed:
Trying setting blame-hang-timeout too

Piedone committed:
Removing useless DOTNET_SYSTEM_NET_SOCKETS_INLINE_COMPLETIONS

Piedone committed:
Using just blame-hang-timeout

Piedone committed:
Dummy change to kick off build

Piedone committed:
Revert "Dummy change to kick off build"

This reverts commit bdbfecdd08b1e7861ca4d55a18e6c7a587c06988.

Piedone committed:
Back to full build workflow

Piedone committed:
Re-add codespell and LGHA branch references

Piedone committed:
Spelling

Piedone committed:
Submodule tools/Lombiq.GitHub.Actions updated

Submodule tools/Lombiq.GitHub.Actions:
    > 555f0518 - Adding cancel-workflow-on-failure to the codespell workflow too
    > b47e1736 - More readable command

Piedone committed:
Intentional typo

Piedone committed:
And again

Piedone committed:
Fixing typo

Piedone committed:
Attempting to fix ZAP report files remaining locked

Piedone committed:
Back to the debug workflow

Piedone committed:
Dummy change to kick off build

Piedone committed:
Revert "Dummy change to kick off build"

This reverts commit 7437e90e6de6ad695a36c3211dc832c07b84e72c.

Piedone committed:
Docs

Piedone committed:
Just running BasicSecurityScanShouldPass

Piedone committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-935

# Conflicts:
#	NuGetTest/src/Lombiq.OSOCE.NuGet.Web/Lombiq.OSOCE.NuGet.Web.csproj
#	NuGetTest/test/Lombiq.OSOCE.NuGet.Tests.UI/Lombiq.OSOCE.NuGet.Tests.UI.csproj
#	src/Modules/Lombiq.HelpfulExtensions

Piedone committed:
Debug code

Piedone committed:
More debug code

Piedone committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-935

# Conflicts:
#	src/Modules/Lombiq.DataTables
#	test/Lombiq.UITestingToolbox

Piedone committed:
Updating debug code

Piedone committed:
More debug code

Piedone committed:
Trying other chmod

Piedone committed:
Fixing chmod

Piedone committed:
Maximum chmod

Piedone committed:
Chmodding the whole temp dir

Piedone committed:
Debug output

Piedone committed:
Updating ZAP to 2.16.0

Piedone committed:
Trying to get process locking ZAP report files

Piedone committed:
Fixing file locking check

Piedone committed:
Trying to stop Docker containers forcefully

Piedone committed:
Alternative docker stop

Piedone committed:
Command syntax

Piedone committed:
Just listing Docker containers

Piedone committed:
Not using anything non-default for the build

Piedone committed:
Trying to not create a test dumb

Piedone committed:
Alternative file locking discovery

Piedone committed:
No sudo

Piedone committed:
Reverting some debug code

Piedone committed:
Trying recursive chmod

Piedone committed:
Fixing debug code

Piedone committed:
Fixing chmod

Piedone committed:
sudo chmod

Piedone committed:
More chmod

Piedone committed:
Removing chmod override

Piedone committed:
Not running clean-up on GitHub

Piedone committed:
Reverting debug code

Piedone committed:
Testing test hang

Piedone committed:
Revert "Testing test hang"

This reverts commit 7752a63e94e0fc830738e3cdeeb5daff3ac8275a.

Piedone committed:
Submodule tools/Lombiq.GitHub.Actions updated

Submodule tools/Lombiq.GitHub.Actions:
    > d1f831e5 - Revert "Temporarily always running hang dump collection"
    > 0382ad1f - Temporarily always running hang dump collection
    > c31a557a - Copy instead of move for blame hang dump files
    > 419a8f4b - Allowing dump files when the build configuration for test execution is set
    > cef5f148 - Revert "Temporarily adding Warpbuilds/action-debugger"
    > 700b6f87 - Temporarily adding Warpbuilds/action-debugger

Piedone committed:
Back to full workflow

Piedone committed:
Spelling

Piedone committed:
Adding blame-hang-timeout to Windows workflow too

Piedone commented on .github/workflows/build-and-test.yml:
```diff
@@ -20,14 +20,14 @@ jobs:
       set-up-elasticsearch: 'true'
       build-create-binary-log: 'true'
-      dotnet-test-process-timeout: 600000
+      blame-hang-timeout: 600000
```
See https://github.com/Lombiq/UI-Testing-Toolbox/issues/228#issuecomment-2605946303

Piedone committed:
Spelling

Piedone committed:
Fixing build error in NuGetTest

Piedone committed:
More efficient EnteringInteractiveModeShouldWait fail

Piedone committed:
Fixing analyzer violations

Piedone committed:
And more

Piedone committed:
Fixing that NavigationWithCustomAdminPrefixShouldWork could break other tests

Piedone committed:
Spelling

Piedone committed:
Making EnteringInteractiveModeShouldWait safer

Piedone committed:
Trying to make EnteringInteractiveModeShouldWait safer

Piedone committed:
Temporarily disabling cancel-in-progress-for-this-pr

Piedone committed:
Dummy change to kick off build

Piedone committed:
Revert "Dummy change to kick off build"

This reverts commit 7132e34e544892196b1b8bba5df86cfc0d37630c.

Piedone committed:
Revert "Temporarily disabling cancel-in-progress-for-this-pr"

This reverts commit f3c9f2ef9a9181ab8cb4ad2ad758f16399353739.

Piedone committed:
Temporarily disabling cancel-in-progress-for-this-pr

Piedone committed:
Making EnteringInteractiveModeShouldWait again more reliable

Piedone committed:
Dummy change to kick off build

Piedone committed:
Revert "Dummy change to kick off build"

This reverts commit 608a0cd5c32ef43a10ec5457a85845008f29baa0.

Piedone committed:
Reapply "Dummy change to kick off build"

This reverts commit 7b01b2436bdca0c8ae39c1b78ad5a5e4f1b2aa59.

Piedone committed:
Fixing workflow

Piedone committed:
Revert "Reapply "Dummy change to kick off build""

This reverts commit 86aacd9a912c36839eaba3d441c2beca6121e73f.

Piedone committed:
Reapply "Reapply "Dummy change to kick off build""

This reverts commit 9a07e9e69331648e488704234ccaec1d515e7242.

Piedone committed:
Revert "Reapply "Reapply "Dummy change to kick off build"""

This reverts commit c2f24b4bc20466f098ea779398bd5466a8884a6f.

Piedone committed:
Submodule test/Lombiq.UITestingToolbox updated

Submodule test/Lombiq.UITestingToolbox:
    > 29aa2fae - Removing EnteringInteractiveModeShouldWait that's impossible to make reliable
    > 1bcb543f - More convenient interactive mode demo

Piedone committed:
Revert "Temporarily disabling cancel-in-progress-for-this-pr"

This reverts commit 8de96fd723a55289762678d4bba4f48fe537f632.

Piedone committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-935

# Conflicts:
#	src/Modules/Lombiq.HelpfulExtensions

Piedone committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-935

Piedone committed:
Revert "Temporarily disabling EmailQuotaShouldBlockEmailsOverLimitAndWarn"

This reverts commit 9fdaa7a4b67b4e7d04a3782bc379986fb6301018.

Piedone committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-935

Piedone committed:
Revert "Temporarily disabling EmailQuotaShouldNotBlockEmailsWhenDifferentHostIsUsedThanOriginalFromConfig too"

This reverts commit b06cfcc35bf797c7a746276f7857f1a8b3d0f50d.

Piedone committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-935

# Conflicts:
#	test/Lombiq.UITestingToolbox

Piedone committed:
Re-enabling SendingTestEmailShouldWork

Piedone committed:
Better name for the Elasticsearch test

Piedone committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-935

# Conflicts:
#	.github/workflows/build-and-test-windows.yml
#	.github/workflows/build-and-test.yml
#	NuGetTest/src/Lombiq.OSOCE.NuGet.Web/Lombiq.OSOCE.NuGet.Web.csproj
#	NuGetTest/test/Lombiq.OSOCE.NuGet.Tests.UI/Lombiq.OSOCE.NuGet.Tests.UI.csproj
#	NuGetTest/test/Lombiq.OSOCE.NuGet.Tests/Lombiq.OSOCE.NuGet.Tests.csproj
#	src/Libraries/Lombiq.HelpfulLibraries
#	src/Lombiq.OSOCE.Web/Lombiq.OSOCE.Web.csproj
#	src/Modules/Lombiq.ChartJs
#	src/Modules/Lombiq.ContentEditors
#	src/Modules/Lombiq.DataTables
#	src/Modules/Lombiq.Hosting.Tenants
#	src/Modules/Lombiq.LoginAsAnybody
#	src/Modules/Lombiq.VueJs
#	src/Themes/Lombiq.BaseTheme
#	src/Utilities/Lombiq.NodeJs.Extensions
#	test/Lombiq.UITestingToolbox
#	tools/Lombiq.GitHub.Actions

Piedone committed:
Fixing package versions after merge

Piedone committed:
Updating xUnit packages

Piedone committed:
Fixing analyzer violations

Piedone committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-935

# Conflicts:
#	src/Utilities/Lombiq.NodeJs.Extensions
#	test/Lombiq.UITestingToolbox

Piedone committed:
Better HTML validation assertion in NuGetTest

Piedone committed:
Submodules src/Libraries/Lombiq.HelpfulLibraries, src/Modules/Lombiq.Hosting.Tenants, test/Lombiq.UITestingToolbox updated

Submodule src/Libraries/Lombiq.HelpfulLibraries:
    > bb1c2ac - Fixing analyzer violations

Submodule src/Modules/Lombiq.Hosting.Tenants:
    > 4ab4faf - Simplified DoWithRetriesOrFailAsync call

Submodule test/Lombiq.UITestingToolbox:
    > b1d3507e - CancellationToken galore

Piedone committed:
Submodules src/Libraries/Lombiq.HelpfulLibraries, test/Lombiq.UITestingToolbox updated

Submodule src/Libraries/Lombiq.HelpfulLibraries:
    > bbd17b5 - Fixing test project

Submodule test/Lombiq.UITestingToolbox:
    > 29e27d31 - Fixing exception when disposing AzureBlobStorageManager
    > f37389f6 - Fixing that double disposing ZapManager and SmtpService caused exception
    > ec45fcd6 - Suppressing breaking changes

Piedone committed:
Submodule src/Libraries/Lombiq.HelpfulLibraries updated

Submodule src/Libraries/Lombiq.HelpfulLibraries:
    > 6b8c760 - And fixing the test project even more

Piedone committed:
Updating NuGetTest

Piedone committed:
NuGetTest updates were actually missing

Piedone committed:
Updating xUnit in NuGetTest too

Piedone committed:
Dummy change to kick off build

Piedone committed:
Revert "Dummy change to kick off build"

This reverts commit 46f9152f6e053225a84920f980f04d9c5720ad62.

Piedone committed:
Simplification

Piedone committed:
More readable response and browser log in UI test output

Piedone committed:
Submodule test/Lombiq.UITestingToolbox updated

Submodule test/Lombiq.UITestingToolbox:
    > 72e23fa2 - Updating CompatibilitySuppressions with irrelevant extensions class rename

Piedone committed:
Merge remote-tracking branch 'origin/issue/OSOE-1001' into issue/OSOE-935

# Conflicts:
#	.github/workflows/build-and-test.yml
#	src/Utilities/Lombiq.NodeJs.Extensions
#	tools/Lombiq.GitHub.Actions

Piedone committed:
Re-publishing NE NuGet

Piedone committed:
Adding xUnit marker interface

Piedone committed:
Submodule test/Lombiq.UITestingToolbox updated

Submodule test/Lombiq.UITestingToolbox:
    > 292c13aa - Revert "Pointing GHA to issue branch"
    > ff1aa673 - Pointing GHA to issue branch

Piedone committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-935

# Conflicts:
#	tools/Lombiq.GitHub.Actions

Piedone committed:
Merging submodules

Piedone committed:
Updating Lombiq.Tests.UI reference to latest alpha

Piedone committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-935

# Conflicts:
#	.github/workflows/build-and-test-windows.yml
#	.github/workflows/build-and-test.yml

Piedone committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-935

# Conflicts:
#	.github/workflows/build-and-test-windows.yml
#	.github/workflows/build-and-test.yml

dministro reviewed (changes requested):
Merge from dev please.

Piedone committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-935

# Conflicts:
#	src/Modules/Lombiq.HelpfulExtensions
#	src/Modules/Lombiq.Hosting.Tenants
#	test/Lombiq.UITestingToolbox
#	tools/Lombiq.GitHub.Actions

Piedone committed:
Submodule test/Lombiq.UITestingToolbox updated

Submodule test/Lombiq.UITestingToolbox:
    > eacb5c0c - Code styling
    > e16e8970 - Removing duplicated await

dministro committed:
Updating branch references
Updating submodules
